sale_order_secondary_unit_price/models/sale_order.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2019 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.addons import decimal_precision as dp
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    secondary_unit_price = fields.Float(
+        'Secondary Unit Price', 
+        digits=dp.get_precision('Product Price')
+    )
+
+    @api.onchange('secondary_unit_price', 'price_unit')
+    def onchange_unit_price(self):
+        if not self.secondary_uom_id:
+            return
+        factor = self.secondary_uom_id.factor * \
+            self.product_uom.factor
+        self.price_unit = self.secondary_unit_price / factor
```

Comment:
@nakometal We may want to mention the intention here that `price_unit` is practically not editable when `secondary_uom_id` is set. I may get confused if there is no explanation.

Reply:
@yostashiro 
Thanks. but I think not necessary add a comment because I'll change XML to lock the price_unit and delete on_change for price_unit.